open street map layer

markjdugger opened the pull request:
Demonstrate use of open street map as a basemap.

markjdugger committed:
created OpenStreetMapLayerSample using OPEN_STREET_MAP basemap

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java:
```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2017 Esri.
```
need to change to 2018

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java:
```diff
@@ -0,0 +1,84 @@
+      stage.setHeight(700);
+      stage.setScene(scene);
+      stage.show();
+
```
remove new line

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java:
```diff
@@ -0,0 +1,84 @@
+      stage.show();
+
+
+      // create an ArcGISMap with an OPEN_STREET_MAP basemap
```
create ~~an ArcGISMap~~ map with an ~~OPEN_STREET_MAP~~ OpenStreetMap basemap

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java:
```diff
@@ -0,0 +1,84 @@
+      // create an ArcGISMap with an OPEN_STREET_MAP basemap
+      ArcGISMap map = new ArcGISMap(Basemap.Type.OPEN_STREET_MAP, 34.056295, -117.195800, 10);
+      
+      // set the map to be displayed in this view
```
"set the map to a map view"

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java:
```diff
@@ -0,0 +1,84 @@
+      mapView.setMap(map);
+	  	  
+      // add the map view to stack pane
+      stackPane.getChildren().addAll(mapView);
```
addAll -> add

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md:
```diff
@@ -0,0 +1,18 @@
+Add the OpenStreetMap layer to a map as a Basemap.
+
+![OpenStreetMap Layer App](openstreetmap-layer.png)
+
```
This file needs to be in html while still having the .md extension. The reason is the sample viewer renders these in a webview as html. Look at the raw text of some of the other sample readmes to get an idea of the format.

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md:
```diff
@@ -0,0 +1,18 @@
+# OpenStreetMap Layer
+Add the OpenStreetMap layer to a map as a Basemap.
+
+![OpenStreetMap Layer App](openstreetmap-layer.png)
```
The image needs to be 800w by 400h.

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md:
```diff
@@ -0,0 +1,18 @@
+
+![OpenStreetMap Layer App](openstreetmap-layer.png)
+
+## How to use the sample
```
You can just omit this section for basic samples like this one.

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md:
```diff
@@ -0,0 +1,18 @@
+Simply run the app.
+
+## How it works
+Instantiate an `ArcGISMap` with a `Basemap.Type.OPEN_STREET_MAP` and add it to a `MapView`. The attribution text that is required by OpenStreetMap is automatically added to the `MapView's` attributionText without any additional code.
```
This is always shown as a list for Java SE. Make this a two step list with the first step to create the map and the second to set it to the map view. The part about the attribution can be omitted.

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md:
```diff
@@ -0,0 +1,18 @@
+## How it works
+Instantiate an `ArcGISMap` with a `Basemap.Type.OPEN_STREET_MAP` and add it to a `MapView`. The attribution text that is required by OpenStreetMap is automatically added to the `MapView's` attributionText without any additional code.
+
+## Relevant API
```
Add `Basemap.Type`

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md:
```diff
@@ -0,0 +1,18 @@
+* MapView
+
+#### Tags
+Layers
```
Add OpenStreetMap

tschie reviewed (changes requested):
Great first start! I've suggested a few changes based on the sample implementation guidelines (I sent you a link separately). Please re-request review after you've made the changes.

markjdugger committed:
Converted README.md to HTML and renamed image

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md:
```diff
@@ -1,18 +1,26 @@
-## How to use the sample
-Simply run the app.
+<img src="OpenStreetMapLayer.png"/>
+<figcaption>OpenStreetMap Layer App</figcaption>
```
Don't need the `figcaption` or `figure`. I also don't think the renderer in the sample viewer will handle it.

markjdugger committed:
Recaptured OpenStreetMapLayer.png and lightly edited README.md

markjdugger committed:
Update OpenStreetMapLayerSample.java

changed copyright to 2018

markjdugger committed:
Update OpenStreetMapLayerSample.java

removed extra newline

markjdugger committed:
Update OpenStreetMapLayerSample.java

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/OpenStreetMapLayerSample.java:
```diff
@@ -44,7 +44,7 @@ public void start(Stage stage) {
       stage.show();
 
-      // create an ArcGISMap with an OPEN_STREET_MAP basemap
+      // create a map with an OpenSteetMap basemap
```
typo Steet

markjdugger committed:
Changed call from addAll(mapView) to stackPane.getChildren().add(mapView)
Corrected typos in code comments
Standardized README text to match existing READMEs in Samples repo

tschie commented on src/main/java/com/esri/samples/ogc/open_street_map_layer/README.md:
```diff
@@ -0,0 +1,25 @@
+   <li>MapView</li>
+</ul>
+
+<h4>Tags</h4>
```
use h2 tag here

tschie reviewed (approved):
Looks good! Only one more thing to change (the `h2` tag in the readme). Feel free to merge after that.

markjdugger committed:
changed Tags heading to <H2>
